src/module/ModuleTypes.jl

```diff
@@ -4,21 +4,27 @@
 #
 ###############################################################################
 
-const FreeModID = Dict{Tuple{PolyRing, Int}, Module}()
+const FreeModID = Dict{Tuple{Union{PolyRing,PluralRing}, Int}, Module}()
 
-mutable struct FreeMod{T <: Nemo.RingElem} <: Module{T}
-   base_ring::PolyRing
+mutable struct FreeMod{T <: Union{Nemo.RingElem, Singular.spluralg}} <: Module{T}
```

Comment:
How about just changing this to 
```suggestion
mutable struct FreeMod{T <: NCRingElem} <: Module{T}
```
And similar in the definition of `FreeModID`, etc. etc.?

That way we are ready for supporting other rings if we need to, and it doesn't really cost us much if anything.